linlearn/_estimator.py

```diff
@@ -128,6 +132,37 @@ def partial_deriv(j, inner_products, state):
 
             return partial_deriv
 
+    def grad_factory(self):
+        X = self.X
+        y = self.y
+        loss = self.loss
+        deriv_loss = loss.deriv_factory()
+        n_samples = self.n_samples
+
+        if self.fit_intercept:
+
+            @jit(**jit_kwargs)
+            def grad(inner_products, state):
+                gradient = np.zeros(X.shape[1] + 1)
```

Comment:
In order to keep using a generic code for method solve() in the mother Solver class I'll allocate the gradient in the estimator state. If we do this through an "out" parameter we would need to override the solve() method, let me know if you prefer that anyway

Reply:
adding the parameter "out" poses the question of where it should be allocated and since we use the hierarchy fit -> solve -> cycle and do not wish to allocate at each cycle, we would have to allocate in solve() but then we would have to pass this gradient place holder as a parameter to cycle which will be useless in many cases. To avoid this, I'll allocate it in the estimator state

Reply:
Yes, you should allocate it in the estimator state